File: app/src/main/java/com/mmjang/ankihelperrefactor/ui/FieldMapSpinnerListAdapter.java
package com.mmjang.ankihelperrefactor.ui;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Spinner;
import android.widget.TextView;

import com.mmjang.ankihelperrefactor.R;
import com.mmjang.ankihelperrefactor.app.FieldsMapItem;
import com.mmjang.ankihelperrefactor.app.MyApplication;

import java.util.List;

/**
 * Created by liao on 2017/4/28.
 */

public class FieldMapSpinnerListAdapter
        extends RecyclerView.Adapter<FieldMapSpinnerListAdapter.ViewHolder>{
    private List<FieldsMapItem> mFieldsMapItemList;

    static class ViewHolder extends RecyclerView.ViewHolder{
        TextView exportElementName;
        Spinner fieldsSpinner;
        public ViewHolder(View view){
            super(view);
            exportElementName = (TextView) view.findViewById(R.id.tv_export_element);
            fieldsSpinner = (Spinner) view.findViewById(R.id.spinner_fields);
        }
    }

    public FieldMapSpinnerListAdapter(List<FieldsMapItem> itemList){
        mFieldsMapItemList = itemList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType){
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.field_map_item, parent, false);
        ViewHolder holder = new ViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position){

        final FieldsMapItem item = mFieldsMapItemList.get(position);
        holder.exportElementName.setText(item.getField());
        ArrayAdapter<String> arrayAdapter = new ArrayAdapter<String>(
                MyApplication.getContext(),
                android.R.layout.simple_spinner_dropdown_item,
                item.getExportedElementNames()
        );
        holder.fieldsSpinner.setAdapter(arrayAdapter);
        holder.fieldsSpinner.setSelection(item.getSelectedFieldPos());
        holder.fieldsSpinner.setOnItemSelectedListener(
                new AdapterView.OnItemSelectedListener() {
                    @Override
                    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                        item.setSelectedFieldPos(position);
                    }

                    @Override
                    public void onNothingSelected(AdapterView<?> parent) {

                    }
                }
        );
    }

    @Override
    public int getItemCount(){
        return mFieldsMapItemList.size();
    }
}

File: app/src/main/java/com/lmyby/ankiquicker/ui/plan/FieldMapSpinnerListAdapter.kt
package com.lmyby.ankiquicker.ui.plan

import android.app.Activity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Spinner
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.lmyby.ankiquicker.R

/**
 * Adapter for field mapping spinner list
 * Converted to Kotlin as part of Phase 4 adapter migration
 */
class FieldMapSpinnerListAdapter(
    private val mActivity: Activity,
    private val mFieldsMapItemList: List<FieldsMapItem>
) : RecyclerView.Adapter<FieldMapSpinnerListAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val exportElementName: TextView = view.findViewById(R.id.tv_export_element)
        val fieldsSpinner: Spinner = view.findViewById(R.id.spinner_fields)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.field_map_item, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = mFieldsMapItemList[position]
        holder.exportElementName.text = item.field

        val arrayAdapter = ArrayAdapter(
            mActivity,
            android.R.layout.simple_spinner_dropdown_item,
            item.exportedElementNames
        )
        holder.fieldsSpinner.adapter = arrayAdapter
        holder.fieldsSpinner.setSelection(item.selectedFieldPos)
        holder.fieldsSpinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
                item.selectedFieldPos = position
            }

            override fun onNothingSelected(parent: AdapterView<*>?) {
                // No action needed
            }
        }
    }

    override fun getItemCount(): Int = mFieldsMapItemList.size
}
